### PR TITLE
add linear-gradient

### PR DESCRIPTION
**Pull request by phinchik**

### CODE REVIEW

File: client/js/App.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from "react";
 import { Text, View } from "react-native";
+import { typography } from "../../client/js/config/styles";
```
Comment (siddharthparmar7) [inline client/js/App.js]: as asked earlier please remove unnecessary imports.

File: client/js/config/styles.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  purple: "#9F49E0",
+  grey: "#D8D8D8"
+};
+
+export const typography = {
+  baseSize: 16,
+  fontMainH1: "Rubik-Regular",
+  fontMainH2: "Rubik-Medium"
+  // fontAccentH3: "Metric-Regular",
+  // fontButton: "Metric-Medium"
```
Comment (siddharthparmar7) [inline client/js/config/styles.js]: remove commented code if you are not using it.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the requested changes. Also, since you have linked the fonts, make sure that they are working and then make a pull request. Thank you.

File: client/js/screens/SignUp/SignUp.js [hunk 3]
```diff
@@ -1,9 +1,104 @@
...
+    <Fragment>
+      <ImageBackground
+        source={require("../../assets/images/Signup.png")}
+        style={styles.backgroundImage}
+      >
+        <Text style={styles.register}>REGISTER</Text>
+        <Form
+          onSubmit={values => onSubmit(values, createUser)}
+          render={({ handleSubmit, pristine, invalid }) => (
+            <View>
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/SignUp.js]: Use Fragment instead. Can't have View component by itself.

File: client/js/screens/SignUp/SignUp.js [hunk 4]
```diff
@@ -1,9 +1,104 @@
...
+              <Field name="firstname">
+                {({ input, meta }) => (
+                  <View>
+                    <TextInput
+                      {...input}
+                      style={styles.input}
+                      placeholder="First name"
+                      placeholderTextColor="white"
+                    />
+                  </View>
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/SignUp.js]: What if there is an error? Please show the errors and validate your input.

File: client/js/screens/SignUp/SignUp.js [hunk 5]
```diff
@@ -1,9 +1,104 @@
...
 import styles from "./styles";
+import { Form, Field } from "react-final-form";
 
-const Signup = () => {
-  return <Text>I am the Signup Screen</Text>;
+const onSubmit = async (values, createUser) => {
+  const res = await createUser({ variables: values });
+};
+
+const Signup = ({ data }) => {
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/SignUp.js]: there is no data prop being passed from your container. Did you mean to say anything else?

File: client/js/screens/SignUp/SignUpContainer.js [hunk 6]
```diff
@@ -1,10 +1,41 @@
...
+`;
+
+class SignupContainer extends Component {
+  render() {
+    return (
+      <Mutation mutation={SignupMutation}>
+        {(createUser, { data }) => (
+          <View>
+            <Signup signup={createUser} />
+            {console.log("this is it", data)}
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/SignUpContainer.js]: remove the console log.

File: client/js/screens/SignUp/SignUpContainer.js [hunk 7]
```diff
@@ -1,10 +1,41 @@
...
     }
+  }
+`;
+
+class SignupContainer extends Component {
+  render() {
+    return (
+      <Mutation mutation={SignupMutation}>
+        {(createUser, { data }) => (
+          <View>
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/SignUpContainer.js]: use Fragment.

File: client/js/screens/SignUp/SignUpContainer.js [hunk 8]
```diff
@@ -1,10 +1,41 @@
...
+      id
+      email
     }
+  }
+`;
+
+class SignupContainer extends Component {
+  render() {
+    return (
+      <Mutation mutation={SignupMutation}>
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/SignUpContainer.js:28]: how are you passing variables in your Mutation query? Missing variables?

File: client/js/screens/SignUp/styles.js [hunk 9]
```diff
@@ -1,9 +1,44 @@
...
 const styles = StyleSheet.create({
-  container: globalStyles.container
+  container: globalStyles.container,
+  register: {
+    fontSize: 25,
+    fontWeight: "bold",
+    color: "white",
+    marginTop: 30,
+    textAlign: "center",
+    marginBottom: 30
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/styles.js:14]: use flexbox not marginTop and marginBottom with value 30.

File: client/js/screens/SignUp/styles.js [hunk 10]
```diff
@@ -1,9 +1,44 @@
...
+  },
+  input: {
+    color: "white",
+    borderBottomWidth: 1,
+    borderBottomColor: "white",
+    margin: 25,
+    marginTop: 20,
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/styles.js:21]: this is a huge margin. Please use Flexbox instead.

File: client/js/screens/SignUp/styles.js [hunk 11]
```diff
@@ -1,9 +1,44 @@
...
+    fontSize: 16
+  },
+  button: {
+    backgroundColor: "white",
+    height: 45,
+    width: 250,
```
Comment (siddharthparmar7) [inline client/js/screens/SignUp/styles.js:27]: use Dimensions or flexbox, not static height and width.

File: client/js/screens/SignUp/SignUp.js (hunk 3 above)
Comment (siddharthparmar7) [reply client/js/screens/SignUp/SignUp.js]: Remove all the View components that are just wrappers and nothing else and use Fragment instead.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: client/js/App.js (hunk 1 above)
Comment (phinchik) [reply client/js/App.js]: this is updated on my newer commit

File: client/js/config/styles.js (hunk 2 above)
Comment (phinchik) [reply client/js/config/styles.js]: done

File: client/js/screens/SignUp/SignUp.js (hunk 3 above)
Comment (phinchik) [reply client/js/screens/SignUp/SignUp.js]: done

File: client/js/screens/SignUp/SignUp.js (hunk 4 above)
Comment (phinchik) [reply client/js/screens/SignUp/SignUp.js]: done

Comment (phinchik) [reply client/js/screens/SignUp/SignUp.js]: done

File: client/js/screens/SignUp/SignUp.js (hunk 5 above)
Comment (phinchik) [reply client/js/screens/SignUp/SignUp.js]: change it to signup cus that's what im passing from the container

File: client/js/screens/SignUp/SignUpContainer.js (hunk 6 above)
Comment (phinchik) [reply client/js/screens/SignUp/SignUpContainer.js]: done

File: client/js/screens/SignUp/SignUpContainer.js (hunk 7 above)
Comment (phinchik) [reply client/js/screens/SignUp/SignUpContainer.js]: done

File: client/js/screens/SignUp/SignUpContainer.js (hunk 8 above)
Comment (phinchik) [reply client/js/screens/SignUp/SignUpContainer.js:28]: the variable are on the signup.js cause the variables are the textInput

File: client/js/screens/SignUp/styles.js (hunk 9 above)
Comment (phinchik) [reply client/js/screens/SignUp/styles.js:14]: i can't see my page cause i'll be needing vanessa's pr "apollo provider" i'll change it once i can navigate to my screen.

Comment (siddharthparmar7) [reply client/js/screens/SignUp/styles.js:14]: Then how did you work on it?

**Review by siddharthparmar7 — COMMENTED** [review]
Make sure I can actually see what you worked on instead of Text Hello in your app.

**Comment by phinchik** [comment]
i'm trying to load my screen to the app.js cause i dont have navigation yet and it's throwing me error cause i need vanessa's api for me to actually get data from the backend.

**Comment by 10thfloor** [comment]
Closing. CHanges were merged into new branch. @phinchik Don't make new branches when there are change requests. Make changes in the original branch. Thanks!

**Pull request closed without merging**